fix crash neutrinordp missing home env

cocoon opened the pull request:
https://github.com/neutrinolabs/xrdp/issues/719

One possible way to fix the crash is to add a fallback path if no home env exists.

Question is: is / a good path for it or could it cause any harm?
Would be /tmp a better path as fallback?

metalefty commented:
Maybe `/` is ok. FreeRDP uses `/`.

metalefty commented on libfreerdp-utils/file.c:
```diff
@@ -83,6 +83,17 @@ char* freerdp_get_home_path(rdpSettings* settings)
+	/* ... */
+	/* User=root */
+	if(settings->home_path == NULL)
+        {
```
Indent is corrupted.

metalefty commented on libfreerdp-utils/file.c:
```diff
@@ -83,6 +83,17 @@ char* freerdp_get_home_path(rdpSettings* settings)
+	/* User=root */
+	if(settings->home_path == NULL)
+        {
+                settings->home_path = "/";
```
`xstrdup("/");` would be better rather than a raw literal.

See: https://github.com/FreeRDP/FreeRDP/commit/036c8ce47d4400f78c693db63852a5bc95fa2cad

If we change it to `xstrdup("/");`, we're backporting the code from FreeRDP. Maybe copyright notices needed. The main author of FreeRDP Marc-Andre Moreau already appears at the top but this homedir fix is provided by Pawel Jakub Dawidek. We need to record his name at least commit message. Of course in the file will be better.

For example, prefered commit message will be like this.

```
Fix crash neutrinordp when missing HOME env

Imported from: https://github.com/FreeRDP/FreeRDP/commit/036c8ce47d4400f78c693db63852a5bc95fa2cad
Originally fixed by:  Pawel Jakub Dawidek
```

cocoon commented:
changes applied and backported fix from FreeRDP

Commit:
Fix crash neutrinordp when missing HOME env

Imported from: https://github.com/FreeRDP/FreeRDP/commit/036c8ce47d4400f78c693db63852a5bc95fa2cad
Originally fixed by:  Pawel Jakub Dawidek

metalefty commented:
Looks good but I don't have merge permission. @jsorg71 can merge. Let's wait.
